gmso/abc/abstract_site.py

```diff
@@ -45,6 +51,14 @@ class Site(GMSOBase):
 
     label_: str = Field("", description="Label to be assigned to the site")
 
+    residue_index_: Optional[StrictInt] = Field(
+        None, description="Residue index for the site"
+    )
+
+    residue_label_: Optional[StrictStr] = Field(
```

Comment:
Rename it to `residue_name` maybe (Suggested by @justinGilmer). What do you think @bc118?

Reply:
Look up the PDB format naming convention for residues